### PR TITLE
Support node version and env vars in code-only config

### PR DESCRIPTION
**Pull request by ameyakhare**
## Description
Adds support for nodeVersion and envVars in code-only config.

## Test plan
Was able to deploy a task locally with these changes.

### CODE REVIEW

File: pkg/deploy/discover/parser/node/parser.ts [hunk 1]
```diff
@@ -44,8 +50,8 @@ const extractTaskConfigs = (files: string[]): AirplaneConfigs => {
   let taskConfigs: TaskDefWithBuildArgs[] = [];
   let viewConfigs: ViewDef[] = [];
   for (const file of files) {
-    const resolvedPath = path.relative(__dirname, file);
-    const lib = resolvedPath.replace(/.ts$/, "");
+    const relPath = path.relative(__dirname, file);
+    const lib = path.join(path.dirname(relPath), path.parse(relPath).name);
```
Comment (leeweisberger) [inline pkg/deploy/discover/parser/node/parser.ts:54]: Can you explain? this seems unrelated?

**Review by leeweisberger — CHANGES_REQUESTED** [review]

File: pkg/deploy/discover/parser/node/parser.ts (hunk 1 above)
Comment (ameyakhare) [reply pkg/deploy/discover/parser/node/parser.ts]: we actually dont need to do the path manipulation at all since we import from js files, so leaving the extension is fine

**Review by leeweisberger — APPROVED** [review]
